### DISCUSSION TITLE
explanation for is_sufficiently_precise calculation in drive details?

### DISCUSSION THREAD

**Discussion by Kosta-Github** [teslamate-org/teslamate#1017] [Q&A]
**Describe the bug**

I am trying to figure out why the second row does not show the `consumption` and `efficiency` values:

![image](https://user-images.githubusercontent.com/2526664/96259208-92bc9480-0fbd-11eb-8fcb-f2dd42183372.png)

**Expected behavior**

I would have expected that `consumption` and `efficiency` values to be available for both rows.

**How to reproduce it (as minimally and precisely as possible)**:

What is this boolean expression trying to model:

```
duration_min > 1 AND distance > 1 AND ( 
  start_position.usable_battery_level IS NULL OR end_position.usable_battery_level IS NULL	OR
  (end_position.battery_level - end_position.usable_battery_level) = 0 
) as is_sufficiently_precise
```

In my case:

| type | value |
|-----|-------|
| `start_usable_battery_level` | `48` |
| `start_battery_level` | `49` |
| `end_usable_battery_level` | `44` |
| `end_battery_level` | `45` |

Why is it required that `end_usable_battery_level` equals `end_battery_level` to show a value here?

**Comment by Kosta-Github** [comment]
Suggestion: why not just use the avg of both values for the start and end levels?

`diff = (end_battery_level - start_battery_level + end_usable_battery_level - start_usable_battery_level) / 2`

**Comment by BabyYeggie** [comment]
This was added due to the battery showing an increase in range when it warms up. The average or maybe weighted average is a good idea.

See   #190 , #234 , and #257 .
